simulator/src/ledger/dumper.rs

```diff
@@ -119,16 +119,16 @@ fn dump_resources<T: Ledger>(vaults: &HashSet<Vid>, ledger: &T) -> Result<(), Di
                 .map(|symbol| format!(", symbol: \"{}\"", symbol))
                 .unwrap_or(String::new()),
         );
-        if let Supply::NonFungible { ids } = vault.total_supply(Actor::SuperUser).unwrap() {
-            for (inner_last, id) in ids.iter().identify_last() {
-                let nft = ledger.get_nft(resource_address, *id).unwrap();
+        if let Supply::NonFungible { keys } = vault.total_supply(Actor::SuperUser).unwrap() {
+            for (inner_last, key) in keys.iter().identify_last() {
+                let nft = ledger.get_nft(resource_address, key).unwrap();
                 let immutable_data = validate_data(&nft.immutable_data()).unwrap();
                 let mutable_data = validate_data(&nft.mutable_data()).unwrap();
                 println!(
                     "{}  {} NFT {{ id: {}, immutable_data: {}, mutable_data: {} }}",
                     if last { " " } else { "│" },
                     list_item_prefix(inner_last),
-                    id,
+                    key,
                     immutable_data,
                     mutable_data
                 );
```

Comment:
We need to update resim and transaction builder to handle passing of NFT buckets (ideally to support all NFT keys including `u128`).